### ISSUE TITLE
Have you consider a yeoman based setup?

### ISSUE THREAD

**Issue by dellagustin** [jekyll-octopod/jekyll-octopod#64]
Hi!

When I started trying out octopod I was not sure about what it was and what it did.

Based on my experience so far I reached the understanding it does mainly two things for you:

1. [Scafolding](https://en.wikipedia.org/wiki/Scaffold_(programming)#Project_generation) - `octopod setup` created a project for you to generate the website
2. jekyll plugin - handy [Liquid Filters](https://jekyllrb.com/docs/liquid/filters/) to support the project that was created in the first step

I was wondering if 1. could not be replaced by a yeoman generator, which would also provide some parametrization for the generated project.

I don't have a lot of experience with yeoman, aside from using it, but in the future I could try to implement this.

**Comment by haslinger** [comment]
Fancy a bit of background story?

I found jekyll-octopod (named octopod in those days) when I was searching for a podcast hosting plugin for Jekyll. It was pretty much abandoned in those days, just a scaffold, not a gem. I spoke with the former developer, long story short, he handed the project over to me and I update dependencies, fixed bugs, made it a bit more modern looking till it fully served my needs and transformed it into a gem. That was appox. 4 years a ago. Using it for 5 or so podcasts myself up to now.
I neither liked the code style, nor the structure, but it was good enough and a lot less work than starting over from stretch. In those days I was pretty much migrated from Ruby to Elixir already, but I still liked Ruby (and still like it today).

In October '18 I did a bit of recap and wrote sg. down on the state of the project: https://github.com/jekyll-octopod/jekyll-octopod/blob/master/ROADMAP.md 

ad 1) it is a bit of a zombie: the scaffolding (that I reduced as much as I could) vs. the gem aspect of it (and extracting of the theming).
ad 2) that was the main benefit, that is probably still 90% of the original developer's work.

Every now and then I'm thinking of doing something from scratch (maybe even a data base backed solution, as most of the potential users shied away because of the command line). E.g. we had commenting with Disqus (a privacy nightmare) and they are really missing. Self hosting comments with (an option) isn't an option for most of the potential user. 
I would probably do it in Dart right now (that's my main language, currently) or in Elixir and Phoenix (the combo I am best in). Ruby would be only third choice, because of sentimental reasons. 

But it I am several other interesting projects right and and jekyll-octopod still works well enough for me, creates the static site, that is responsive. The rss Feed works with iTunes and all Podcatchers. That's enough for now. Maybe, when I retire ;-)

I did never use yeoman, to be honest, so no strong feelings there.

So maybe - take a step back - think about the whole thing. Maybe you want to
* take it over (the whole project)
* fork it away
* just pick up some ideas and do something completely different: You could have me as a friendly user, as soon, as your thingy does it as good as jekyll-octopod

What are your thoughts?

**Comment by dellagustin** [comment]
> Fancy a bit of background story?
I love some background history 😍 

Thanks a lot.
Some unorganized thoughts on my side, but I will come back when the hurricane is a little bit more under control.

I think I see a beautiful tech friendship being built here 😉

By the way, soon I will be able to show the results of my work using jekyll-octopod, I think you will gonna like it.

**Comment by haslinger** [comment]
Interested in your thoughts, but let them ... what's the word .. mature, breed. Give them time.
Always happy in friendly conversations. Your issue report was definitely in the top ten percent of friendliness.
Take your time, stay safe and we'll talk again :-)

**Comment by dellagustin** [comment]
> Interested in your thoughts, but let them ... what's the word .. mature, breed. Give them time.
Very wise words.

As promised, here is the result of my work based on jekyll-octopod: https://coronablogbr.org/

I teamed up with a public health researcher in the epidemic spread field to produce a podcast in portuguese (out native language), and  blog.

Our objective is to help bring non-alarmistic and curated information to the population.

I managed to put it up in through the weekend, also using your website as reference for some tips on how to customize it after generation.

Thanks! ❤️
